### PR TITLE
App metrics

### PR DESCRIPTION
**Pull request by NoahNsimbe**
#### Summary of Changes (What does this PR do?)
- Adds firebase analytics events to the app.
- Updates app version to 2.0.4

#### Is this change ready to hit production in its current state?
- [x] Yes
- [ ] No

#### Status of maturity (all need to be checked before merging):

- [x] I consider this code done
- [x] I've tested this locally
- [ ] This branch has been LGTM'ed by a reviewer

#### How should this be manually tested?
* Follow instructions in the [Readme file](https://github.com/airqo-platform/AirQo-frontend/tree/MOB-61-app-metrics/mobile#airqo-mobile-app)

#### What are the relevant tickets?
- [MOB-61](https://airqoteam.atlassian.net/browse/MOB-61)

#### Screenshots (optional)

### CODE REVIEW

File: mobile/ios/Runner.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -392,7 +392,7 @@
 					"$(inherited)",
 					"@executable_path/Frameworks",
 				);
-				MARKETING_VERSION = 2.0.3;
+				MARKETING_VERSION = 2.0.4;
```
Comment (NoahNsimbe) [inline mobile/ios/Runner.xcodeproj/project.pbxproj:395]: updating iOS version

File: mobile/lib/models/event.dart [hunk 2]
```diff
@@ -0,0 +1,68 @@
+import 'package:flutter/foundation.dart';
+
+enum AnalyticsEvent {
```
Comment (NoahNsimbe) [inline mobile/lib/models/event.dart:3]: Events enum

File: mobile/lib/models/user_details.dart [hunk 3]
```diff
@@ -2,6 +2,38 @@ import 'package:json_annotation/json_annotation.dart';
 
 part 'user_details.g.dart';
 
+enum titleOptions { ms, mr, undefined }
+
+enum gender { male, female, undefined }
```
Comment (NoahNsimbe) [inline mobile/lib/models/user_details.dart:7]: Using enums for title

File: mobile/lib/screens/kya/kya_final_page.dart [hunk 4]
```diff
@@ -75,7 +75,8 @@ class _KyaFinalPageState extends State<KyaFinalPage> {
   void initState() {
     super.initState();
     _appService = AppService(context);
-    kya = widget.kya..progress = widget.kya.lessons.length;
+    kya = widget.kya
```
Comment (NoahNsimbe) [inline mobile/lib/screens/kya/kya_final_page.dart:78]: bug fix on KYA

File: mobile/lib/services/app_service.dart [hunk 5]
```diff
@@ -379,14 +387,58 @@ class AppService {
...
+        _logNetworkProvider(userDetails),
+        _logGender(userDetails),
+        _logPlatformType(),
+      ]);
     } catch (exception, stackTrace) {
       debugPrint('$exception\n$stackTrace');
     }
   }
 
+  Future<void> _logPlatformType() async {
```
Comment (NoahNsimbe) [inline mobile/lib/services/app_service.dart:404]: Logging phone type

File: mobile/lib/services/app_service.dart [hunk 6]
```diff
@@ -379,14 +387,58 @@ class AppService {
...
+    if (Platform.isAndroid) {
+      await logEvent(AnalyticsEvent.androidUser);
+    } else if (Platform.isIOS) {
+      await logEvent(AnalyticsEvent.iosUser);
+    } else {
+      debugPrint('Unknown Platform');
+    }
+  }
+
+  Future<void> _logGender(UserDetails userDetails) async {
```
Comment (NoahNsimbe) [inline mobile/lib/services/app_service.dart:414]: logging gender

File: mobile/lib/services/app_service.dart [hunk 7]
```diff
@@ -379,14 +387,58 @@ class AppService {
...
+  }
+
+  Future<void> _logFavPlaces() async {
+    var favPlaces = await _dbHelper.getFavouritePlaces();
+    if (favPlaces.length >= 5) {
+      await logEvent(AnalyticsEvent.savesFiveFavorites);
+    }
+  }
+
+  Future<void> _logNetworkProvider(UserDetails userDetails) async {
```
Comment (NoahNsimbe) [inline mobile/lib/services/app_service.dart:431]: logging network provider

File: mobile/pubspec.yaml [hunk 8]
```diff
@@ -14,7 +14,7 @@ publish_to: 'none' # Remove this line if you wish to publish to pub.dev
 # Read more about iOS versioning at
 # https://developer.apple.com/library/archive/documentation/General/Reference/InfoPlistKeyReference/Articles/CoreFoundationKeys.html
 # android test
-version: 2.0.3+20006
+version: 2.0.4+20008
```
Comment (NoahNsimbe) [inline mobile/pubspec.yaml:17]: updating android version

File: mobile/lib/widgets/buttons.dart [hunk 9]
```diff
@@ -222,6 +194,38 @@ Widget onBoardingNotificationIcon() {
   );
 }
 
+Widget proceedAsGuest({required BuildContext context}) {
+  return GestureDetector(
+    onTap: () {
+      CloudAnalytics().logEvent(AnalyticsEvent.browserAsAppGuest, false);
```
Comment (NoahNsimbe) [inline mobile/lib/widgets/buttons.dart:200]: logging anonymous users

File: mobile/lib/services/native_api.dart [hunk 10]
```diff
@@ -412,7 +413,7 @@ class NotificationService {
   Future<bool> allowNotifications() async {
     var enabled = await requestPermission();
     if (enabled) {
-      await _cloudAnalytics.logEvent(AnalyticsEvent.allowNotification);
+      await _cloudAnalytics.logEvent(AnalyticsEvent.allowNotification, true);
```
Comment (NoahNsimbe) [inline mobile/lib/services/native_api.dart:416]: logging notifications

File: mobile/lib/services/native_api.dart [hunk 11]
```diff
@@ -543,7 +544,7 @@ class RateService {
   }
 
   Future<void> _logAppRating() async {
-    await _cloudAnalytics.logEvent(AnalyticsEvent.rateApp);
+    await _cloudAnalytics.logEvent(AnalyticsEvent.rateApp, true);
```
Comment (NoahNsimbe) [inline mobile/lib/services/native_api.dart:547]: logging app rating

File: mobile/lib/services/native_api.dart [hunk 12]
```diff
@@ -774,7 +775,8 @@ class ShareService {
     }
 
     if (value >= 5) {
-      await _cloudAnalytics.logEvent(AnalyticsEvent.shareAirQualityInformation);
+      await _cloudAnalytics.logEvent(
```
Comment (NoahNsimbe) [inline mobile/lib/services/native_api.dart:778]: logging event when a user shares air quality information

File: mobile/pubspec.yaml (hunk 8 above)
Comment (Baalmart) [reply mobile/pubspec.yaml:17]: Curious about this android version update. Any considerations for backward compatibility?

**Review by Baalmart — APPROVED** [review]
LGTM to me. Just left one comment for your review

File: mobile/pubspec.yaml (hunk 8 above)
Comment (NoahNsimbe) [reply mobile/pubspec.yaml:17]: If I have gotten the question right......Backward compatibility is handled within the application's logic (SqLite database setup, APIs consumed, etc)

Here we are updating two things
1. the version 2.0.4
2. the build number 20008

The build number must always be incremented for every build. So a version can have multiple build numbers. You can think of the version as similar to our docker image versioning though rolling back is close to impossible for mobile apps.

@Baalmart

**Pull request merged**